h2o-core/src/main/java/water/ExternalFrameUtils.java

```diff
@@ -46,57 +52,87 @@ public static ByteChannel getConnection(String ipPort) throws IOException{
         return getConnection(split[0], Integer.parseInt(split[1]));
     }
 
-    public static byte[] vecTypesFromExpectedTypes(byte[] expectedTypes){
-        byte[] vecTypes = new byte[expectedTypes.length];
-        for (int i = 0; i < expectedTypes.length; i++) {
-            switch (expectedTypes[i]){
-                case EXPECTED_BOOL: vecTypes[i] = Vec.T_NUM; break;
-                case EXPECTED_BYTE: vecTypes[i] = Vec.T_NUM; break;
-                case EXPECTED_CHAR: vecTypes[i] = Vec.T_NUM; break;
-                case EXPECTED_SHORT: vecTypes[i] = Vec.T_NUM; break;
-                case EXPECTED_INT: vecTypes[i] = Vec.T_NUM; break;
-                case EXPECTED_LONG: vecTypes[i] = Vec.T_NUM; break;
-                case EXPECTED_FLOAT: vecTypes[i] = Vec.T_NUM; break;
-                case EXPECTED_DOUBLE: vecTypes[i] = Vec.T_NUM; break;
-                case EXPECTED_STRING: vecTypes[i] = Vec.T_STR; break;
-                case EXPECTED_TIMESTAMP: vecTypes[i] = Vec.T_TIME; break;
-                default: throw new IllegalArgumentException("Unknown expected type: "+expectedTypes[i]);
+    public static byte[] vecTypesFromExpectedTypes(byte[] expectedTypes, int[] vecElemSizes){
+        int size = expectedTypes.length;
+        if(vecElemSizes != null && vecElemSizes.length == 0){
```

Comment:
This is wrong as well, the if condition should be valid only if vecElemSizes.length > 0

Furthermore, it should pass an empty array.